src/core/datastructures/camera.cpp

```diff
@@ -271,105 +313,90 @@ bool SkewedPerspectiveCamera::update(const Camera* source) {
     }
 
-    widthProp->setVisible(true);
-    widthCallbackHolder_ = widthProp->onChangeScoped([this, widthProp]() {
-        // Left, right, bottom, top view volume
-        const float oldWidth{frustum_.y - frustum_.x};
-        const float oldHeight{frustum_.w - frustum_.z};
-        auto aspect = oldWidth / oldHeight;
-        const float width = widthProp->get();
-        setFrustum({-width / 2.0f, width / 2.0f, -width / 2.0f / aspect, +width / 2.0f / aspect});
-    });
+    fovProp->setVisible(true);
+    fovCallbackHolder_ = fovProp->onChangeScoped([this, fovProp]() { setFovy(fovProp->get()); });
 
     if (offsetProp) {
-        const vec2 offset = offsetProp->get();
-        setFrustumOffset(offset);
+        setOffset(offsetProp->get());
     } else {
-        offsetProp = new FloatVec2Property("seperation", "Separation", vec2(0.0f), vec2(-10.0f),
+        offsetProp = new FloatVec2Property("separation", "Separation", vec2(0.0f), vec2(-10.0f),
                                            vec2(10.0f), vec2(0.01f));
-        comp->insertProperty(comp->size() - 1, offsetProp, true);
         offsetProp->setSerializationMode(PropertySerializationMode::All);
+        offsetProp->setCurrentStateAsDefault();
+        comp->insertProperty(comp->size() - 1, offsetProp, true);
     }
 
     offsetProp->setVisible(true);
-    offsetCallbackHolder_ = offsetProp->onChangeScoped([this, offsetProp, widthProp]() {
-        const vec2 offset = offsetProp->get();
-        const float oldWidth{frustum_.y - frustum_.x};
-        const float oldHeight{frustum_.w - frustum_.z};
-        auto aspect = oldWidth / oldHeight;
-        const float width = widthProp->get();
-        setFrustumOffset(offset);
-        setFrustum({-width / 2.0f, width / 2.0f, -width / 2.0f / aspect, +width / 2.0f / aspect});
-    });
+    offsetCallbackHolder_ =
+        offsetProp->onChangeScoped([this, offsetProp]() { setOffset(offsetProp->get()); });
 }
 
 bool operator==(const SkewedPerspectiveCamera& lhs, const SkewedPerspectiveCamera& rhs) {
-    return !(lhs.equalTo(rhs) | glm::any(glm::notEqual(lhs.frustum_, rhs.frustum_)) |
-             glm::any(glm::notEqual(lhs.frustumSkewOffset_, rhs.frustumSkewOffset_)));
+    return lhs.equalTo(rhs) && lhs.fovy_ == rhs.fovy_ && lhs.aspectRatio_ == rhs.aspectRatio_ &&
+           glm::all(glm::equal(lhs.offset_, rhs.offset_));
 }
 
 bool operator!=(const SkewedPerspectiveCamera& lhs, const SkewedPerspectiveCamera& rhs) {
-    return (lhs.equalTo(rhs) | glm::any(glm::notEqual(lhs.frustum_, rhs.frustum_)) |
-            glm::any(glm::notEqual(lhs.frustumSkewOffset_, rhs.frustumSkewOffset_)));
-}
-
-float SkewedPerspectiveCamera::getAspectRatio() const {
-    // Left, right, bottom, top view volume
-    const float width{frustum_.y - frustum_.x};
-    const float height{frustum_.w - frustum_.z};
-    return width / height;
+    return (lhs.equalTo(rhs) || (lhs.fovy_ != rhs.fovy_) ||
+            (lhs.aspectRatio_ != rhs.aspectRatio_) |
```

Comment:
why not `!operator==()` ?

Reply:
ahh did I not change that one... changed the others...